Analyze a small, publicly available dataset using Python's NumPy or Pandas libraries.                                                                                                
1. Explore the data:

Print the first few rows using data.head().
Check the data types of each column using data.info().
Get summary statistics (mean, median, standard deviation, etc.) using data.describe().







In [1]:
#Importing Libraries
import pandas as pd
import numpy as np

In [2]:
# Loading the Netflix DataSet via url link

url = "https://raw.githubusercontent.com/allenkong221/netflix-titles-dataset/main/netflix_titles.csv"
df = pd.read_csv(url)


In [3]:
# Data Exploration
df.head()

,show_id,type,title,director,cast,country,date_added,release_year,rating,duration,listed_in,description
0,81145628,Movie,Norm of the North: King Sized Adventure,"Richard Finn, Tim Maltby","Alan Marriott, Andrew Toth, Brian Dobson, Cole...","United States, India, South Korea, China","September 9, 2019",2019,TV-PG,90 min,"Children & Family Movies, Comedies",Before planning an awesome wedding for his gra...
1,80117401,Movie,Jandino: Whatever it Takes,NaN,Jandino Asporaat,United Kingdom,"September 9, 2016",2016,TV-MA,94 min,Stand-Up Comedy,Jandino Asporaat riffs on the challenges of ra...
2,70234439,TV Show,Transformers Prime,NaN,"Peter Cullen, Sumalee Montano, Frank Welker, J...",United States,"September 8, 2018",2013,TV-Y7-FV,1 Season,Kids' TV,"With the help of three human allies, the Autob..."
3,80058654,TV Show,Transformers: Robots in Disguise,NaN,"Will Friedle, Darren Criss, Constance Zimmer, ...",United States,"September 8, 2018",2016,TV-Y7,1 Season,Kids' TV,When a prison ship crash unleashes hundreds of...
4,80125979,Movie,#realityhigh,Fernando Lebrija,"Nesta Cooper, Kate Walsh, John Michael Higgins...",United States,"September 8, 2017",2017,TV-14,99 min,Comedies,When nerdy high schooler Dani finally attracts...


In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6234 entries, 0 to 6233
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       6234 non-null   int64 
 1   type          6234 non-null   object
 2   title         6234 non-null   object
 3   director      4265 non-null   object
 4   cast          5664 non-null   object
 5   country       5758 non-null   object
 6   date_added    6223 non-null   object
 7   release_year  6234 non-null   int64 
 8   rating        6224 non-null   object
 9   duration      6234 non-null   object
 10  listed_in     6234 non-null   object
 11  description   6234 non-null   object
dtypes: int64(2), object(10)
memory usage: 584.6+ KB


Observations
- There are 6234 records in the dataset
- There are missing values in the director, cast and country columns
- Most of the columns has a text datatype asides the show_id and release_year columns

In [6]:

df.describe() #This will only output results for numerical features/variables/columns

,show_id,release_year
count,6.234000e+03,6234.00000
mean,7.670368e+07,2013.35932
std,1.094296e+07,8.81162
min,2.477470e+05,1925.00000
25%,8.003580e+07,2013.00000
50%,8.016337e+07,2016.00000
75%,8.024489e+07,2018.00000
max,8.123573e+07,2020.00000


The most average release year is 2013


2. Calculate basic statistics:

Calculate the mean, median, mode, and standard deviation for each numerical feature.
Calculate the correlation between different features using data.corr().





In [8]:
# re-confirm the numerial features
df.select_dtypes(include="number").columns


Index(['show_id', 'release_year'], dtype='object')

In [9]:
df.release_year.mean()


np.float64(2013.3593198588387)

In [10]:
df.release_year.median()


np.float64(2016.0)

In [11]:
df.release_year.mode()


0    2018
Name: release_year, dtype: int64

In [12]:
df.release_year.std()


np.float64(8.811620351521556)

In [13]:
df.corr(numeric_only=True)


,show_id,release_year
show_id,1.000000,0.566577
release_year,0.566577,1.000000


 Here, we can see that the correlation is 1 for release_year and there is no relevant numeric feature to compare this with.
 Therefore, feature engineering is required for 'duration' so as to determine actual correlation.
 This is outside this module's scope and will be revisited down the line.